src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -115,3 +126,83 @@ export const handleStreamEncryptedResponses: RequestHandler<
       return res.end()
     })
 }
+
+/**
+ * Handler for GET /:formId/adminform/submissions
+ *
+ * @returns 200 with encrypted submission data response
+ * @returns 404 if submissionId cannot be found in the database
+ * @returns 500 if any errors occurs in database query or generating signed URL
+ */
+export const handleGetEncryptedResponse: RequestHandler<
+  { formId: string },
+  unknown,
+  unknown,
+  { submissionId: string }
+> = async (req, res) => {
+  const { submissionId } = req.query
+  const { formId } = req.params
+
+  const logMeta = {
+    action: 'handleGetEncryptedResponse',
+    submissionId,
+    formId,
+  }
+
+  // Step 1: Retrieve submission.
+  const submissionResult = await getEncryptedSubmissionData(
+    formId,
+    submissionId,
+  )
+
+  if (submissionResult.isErr()) {
+    logger.error({
+      message: 'Failure retrieving encrypted submission from database',
+      meta: logMeta,
+      error: submissionResult.error,
+    })
+
+    const { statusCode, errorMessage } = mapRouteError(submissionResult.error)
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
+
+  // Step 2: Retrieve presigned URLs for attachments.
+  const submission = submissionResult.value
+  // Remaining login duration in seconds.
+  const urlExpiry = (req.session?.cookie.maxAge ?? 0) / 1000
```

Comment:
hm, the default 0 wasn't there before right? how does this change the behaviour if req.session.cookie.maxAge was undefined?

Reply:
No la, got session means got cookie + maxAge liow. This is just for linter (and also safety).

Reply:
But hor this is set in `config#cookieSettings`, should we make it more obvious just in case and use it from config directly instead of session?